src/index.js

```diff
@@ -9,12 +9,11 @@
 * This line should be called before any other event listeners in your application
 **************************/
 addEventListener('fetch', event => {
-  event.waitUntil(logRequest(event));
+  event.respondWith(logRequest(event));
```

Comment:
Just wanted to add comment here for future reference. cloudflare documentation does say worker could use any of respondWith or  waitUntil.  True, we should respond with response to the client without the delay. Actually that is what we do inside the logRequest fcn.. We do use waitUntil to actually send event to Moesif, but return response immediately.  So we are not delaying the actual response though we use respondWith in fetch eventHandler at high level
```
} else {
    const after = new Date();
    const txId = request.headers.get(TRANSACTION_ID_HEADER) || uuid4();
  
    event.waitUntil(
      tryTrackRequest(
        event,
        request,
        (response ? response : new EmptyResponse()),
        before,
        after,
        txId,
        requestBody,
        userId, 
        companyId
      )
    );
  
    if (!disableTransactionId && response && response.body) {
      const responseClone = new Response(response.body, response);
      responseClone.headers.set(TRANSACTION_ID_HEADER, txId);
      return responseClone;
    } else {
      return response;
    }
```